java/searching/binary_search.java

```diff
@@ -0,0 +1,40 @@
+public class binary_search{
+
+    static int binarySearch(int[] arr1, int key){
+    int start = 0;
+    int end = arr1.length - 1;
+
+    while (start <= end) {
+        int middle = (int)Math.floor((start + end) / 2);
+        System.out.println("middle: "+middle);
+
+        if (arr1[middle] == key) {
+            // found the key
+            return middle;
+        } else if (arr1[middle] < key) {
+            // continue searching to the right
+            start = middle + 1;
+        } else {
+            // search searching to the left
+            end = middle - 1;
+        }
+    }
+	// key wasn't found
+    return -1;
+    }
+
+    public static void main(String args[])
+    {
+        int arr1[] = {10,20,30,40,50};
```

Comment:
C style declaration of array

Reply:
It works fine... Do I need to change it??